clarity/src/vm/analysis/type_checker/mod.rs

```diff
@@ -296,9 +301,68 @@ impl FunctionType {
         }
     }
 
+    pub fn principal_to_callable_type(&self, value: &Value, depth: u8) -> TypeResult {
```

Comment:
Can you add a docstring here?  It wasn't clear until I got to the end of this method that this is a no-op for non-principal atomic types like `uint`.

Reply:
Added in c826a7a6d38bd1d98127a10f2b68fe03d6b6aa29